train.py

```diff
@@ -398,6 +404,17 @@ def loss_fn(pred, labels):
             optimizers.step()
             lr_schedulers.step()
 
+            if job_config.training.precompute_float8_dynamic_scale_for_fsdp:
```

Comment:
discussed offline: can refactor to make it simpler

Reply:
removed assertion on `enable_fp8_linear/enable_fsdp_fp8_all_gather=True`